### DISCUSSION TITLE
Disable retries after termination (when grace period is exceeded)

### DISCUSSION THREAD

**Discussion by collimarco** [sidekiq/sidekiq#5229] [Q&A]
I have a particular job that should not be retried when it fails:

```
sidekiq_options retry: 0
```

Now I read that Sidekiq has graceful termination after SIGTERM. However, what happens if the job does not finish within the grace period (e.g. 30s)? Normally it will be rescheduled.

The question is: if I set `retry: 0`, does that prevent the job to be rescheduled even in this case?

**Comment by mperham** [comment]
Nope, it does not. A normal job retry is different from re-enqueueing due to shutdown.

**Reply by collimarco** [reply]
Thanks for the clarification! I feared that. Is there any other option to prevent re-enqueueing even in the case of forced termination?

**Reply by mperham** [reply]
“ Just remember that Sidekiq will execute your job at least once, not exactly once.”

**Reply by collimarco** [reply]
@mperham Yes, I know the best practices (e.g. idempotence), and usually apply that, but in this specific case it's not possible.

Is there at least a way to **increase the grace period of Sidekiq**? For example, If I set a `terminationGracePeriodSeconds` to 120 seconds for the pods in Kubernetes, then how can I raise the grace period also in Sidekiq? (i.e. give more time to jobs to complete before they are sent back in queue)

**Comment by mperham** [comment]
Nope, see best practice #2: https://github.com/mperham/sidekiq/wiki/Best-Practices#2-make-your-job-idempotent-and-transactional

**Comment by mperham** [comment]
This discussion UI is terrible. Sorry for the mess!

**Comment by mperham** [comment]
Use sidekiq —help to see command arguments, including shutdown timeout.
